### DISCUSSION TITLE
Templater error: Template parsing error - Template hotkey running a second time

### DISCUSSION THREAD

**Discussion by prattatx** [SilentVoid13/Templater#1499] [Help]
In the Template Hotkeys section of Templater, I set a hotkey to launch a template. If that template file already exists, I get a "Template parsing error, aborting. Destination file already exists". 

My ideal use case: Creating a daily journal. I click the hot key to write morning reflection. I may or may not leave that file open in my workspace, and I may do it mobile or desktop. I go about my day. In the evening, I reflect on the day. I press the hotkey, and it shifts to or re-opens the same file.

**Comment by Zachatoo** [comment]
I would do something like this. Replace the `"path/to/note"` with however you are generating your daily journal name (maybe `tp.date.now("YYYY-MM-DD")`? Replace the `Templates/prattatx` with the path to the name of your current template for your daily journal. Then you set this new template as the hotkey instead of your daily journal template and it will manage either creating or opening today's daily journal.

```js
<%*
const path = "path/to/note";
const file = await tp.file.find_tfile(path);
if (!file) {
  const template = tp.file.find_tfile("Templates/prattatx");
  await tp.file.create_new(template, path, true);
} else {
  app.workspace.getLeaf(false).openFile(file);
}
-%>
```
